src/library/scala/collection/immutable/Vector.scala

```diff
@@ -120,12 +120,13 @@ sealed abstract class Vector[+A] private[immutable] (private[immutable] final va
 
   override def iterableFactory: SeqFactory[Vector] = Vector
 
-  override final def length: Int =
-    if(this.isInstanceOf[BigVector[_]]) this.asInstanceOf[BigVector[_]].length0
-    else prefix1.length
+  override final def length: Int = this match {
+    case value: BigVector[_] => value.length0
+    case _ => prefix1.length
+  }
 
   override final def iterator: Iterator[A] =
-    if(this.isInstanceOf[Vector0.type]) Vector.emptyIterator
+    if(this eq Vector0) Vector.emptyIterator
```

Comment:
comparing the instance requires getting the instance first, a type check is faster

Reply:
Uuuh.  That's a nice one!